### DISCUSSION TITLE
Light/dark selector

### DISCUSSION THREAD

**Discussion by gitpdpo** [GetPublii/Publii#1818] [Ideas]
Hi
Publii sites deserve a selector to switch between dark and light themes as required.

Now the theme is applied automatically depending on the browser settings.
Many sites have a small icon (sun/moon) to manually select the preferred setting.
:-)

**Comment by bobmitro** [comment]
How many....:)

**Reply by michaelfluegge** [reply]
according to chatGTP it's about 10% of all sites ...

**Comment by AKickIndieNutz** [comment]
I guess you could implement something yourself if it isn't something that will be developed by the team, tutorial at https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_toggle_dark_mode

**Comment by Nevarro** [comment]
Does anyone have a suggestion for adding this functionality to the stock Terminal theme? I don't really like that it's dark all the time, but I'd love to add an automated selection and/or a toggle. I'm not really familiar with CSS though.

**Reply by candidexmedia** [reply]
For automated: you'll want to check out the `prefers-color-scheme` media query: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

For the toggle only: see the previous comment with the w3 schools code sample

For a bit of both, see this code sample: https://codepen.io/marcop135/pen/mdJexrQ

Go into your dev tools to observe what happens in those code samples. For the most part, a class is added or removed from certain elements when the toggle button is pressed.

To learn more about HTML, CSS, JS, I recommend:
- [FreeCodeCamp](https://www.freecodecamp.org/learn/2022/responsive-web-design)
- [Mimo](https://mimo.org/)
- [Plan, Design, Code](https://www.superhi.com/catalog/plan-design-and-code-your-first-website)

**Reply by candidexmedia** [reply]
Update: I figured out how to create a toggle for the Simple theme, and made a gist: https://gist.github.com/candidexmedia/e1dc9fb0ce85eddeadc236d6b21e25cd

Try it out with your theme. 😊

**Comment by nqnh** [comment]
Tell me which theme you are using, I am doing this job and it will be finished in one or two days.
I am not familar with plugins but I can change the theme to satisfied you.

**Reply by gitpdpo** [reply]
My theme is LIGNES 2.1.0.0
https://marketplace.getpublii.com/themes/lignes/

**Reply by nqnh** [reply]
It's a pity I cann't download and modify your theme since I have only the free themes.
You'd better ask for the official man such as Bob.
In fact I have finished my site. It is working just as your requirement.

**Reply by gitpdpo** [reply]
Wow. I will be really happy to add this to my theme.
please suggest me how to do it.
I can share the theme but I suppose that it will breach terms and conditions.

maybe Bob can authorize us? @bobmitro

**Reply by bobmitro** [reply]
> I can share the theme but I suppose that it will breach terms and conditions.
> 
> maybe Bob can authorize us? @bobmitro

Unfortunately, sharing the premium products without proper authorization constitutes a clear violation of our terms and conditions. It undermines the integrity of our product and the efforts that go into developing and maintaining it. Unauthorized distribution not only breaches our licensing agreement but also harms our business and community of paying users.

Thank you for your understanding.
